### PR TITLE
Sprint 26 odim 3318 - Code changes for ODIM to support Redis cluster with sentinel

### PR DESCRIPTION
**Pull request by hemantha-beecherla**
PR for - Code changes for ODIM to support Redis cluster with sentinel ODIM-Project#72.

### CODE REVIEW

File: lib-persistence-manager/persistencemgr/redis.go [hunk 1]
```diff
@@ -16,63 +16,241 @@
...
+var redisExtCalls RedisExternalCalls
+
+func init() {
+	redisExtCalls = redisExtCallsImp{}
+}
+
+func sentinelNewClient(host string) *redisSentinel.SentinelClient {
+//	sentinelPort := config.Data.DBConf.RedisSentinelPort
+	rdb := redisExtCalls.newSentinelClient(&redisSentinel.Options{
+		Addr:     host + ":" + "26379",
```
Comment (irappa-pattar) [inline lib-persistence-manager/persistencemgr/redis.go]: is this default port? if so make it constant

File: lib-persistence-manager/persistencemgr/redis.go [hunk 2]
```diff
@@ -16,63 +16,241 @@
...
+
+	return masterIP, masterPort
+}
+
+//resetDBWriteConection is used to reset the WriteConnection Pool (inmemory / OnDisk).
+func resetDBWriteConection(dbFlag DbType) {
+        switch dbFlag {
+        case InMemory:
+                if config.Data.DBConf.RedisHAEnabled {
+//                      time.Sleep(8*time.Second)
```
Comment (irappa-pattar) [inline lib-persistence-manager/persistencemgr/redis.go]: remove commented code

File: lib-persistence-manager/persistencemgr/redis.go [hunk 3]
```diff
@@ -16,63 +16,241 @@
...
+                                atomic.StorePointer((*unsafe.Pointer)(unsafe.Pointer(&inMemDBConnPool.WritePool)), unsafe.Pointer(writePool))
+                                inMemDBConnPool.MasterIP = currentMasterIP
+                                inMemDBConnPool.PoolUpdatedTime = time.Now()
+                                inMemDBConnPool.Mux.Unlock()
+                        }
+                }
+                return
+        case OnDisk:
+                if config.Data.DBConf.RedisHAEnabled {
+//                      time.Sleep(8*time.Second)
```
Comment (irappa-pattar) [inline lib-persistence-manager/persistencemgr/redis.go]: remove commented code

File: lib-persistence-manager/persistencemgr/redis.go [hunk 4]
```diff
@@ -244,33 +438,40 @@ func (p *ConnPool) FilterSearch(table, key, path string) (interface{}, *errors.E
...
 		delkey := string(data.([]uint8))
-		_, err := c.Do("DEL", delkey)
+		_, err := writeConn.Do("DEL", delkey)
 		if err != nil {
 			if errs, aye := isDbConnectError(err); aye {
+				atomic.StorePointer((*unsafe.Pointer)(unsafe.Pointer(&p.WritePool)), nil)
 				return errs
 			}
-			return errors.PackError(errors.UndefinedErrorType, errorCollectingData, err)
+//			return errors.PackError(errors.UndefinedErrorType, errorCollectingData, err)
```
Comment (irappa-pattar) [inline lib-persistence-manager/persistencemgr/redis.go]: remove commented code

File: lib-utilities/go.mod [hunk 5]
```diff
@@ -9,3 +9,4 @@ require (
 	github.com/kataras/iris/v12 v12.1.9-0.20200616210209-a85c83b70ad0
 	github.com/micro/go-micro v1.13.2
 )
+replace github.com/ODIM-Project/ODIM/lib-persistence-manager v0.0.0-20200727095517-f2a4bd1bdef7 => ../lib-persistence-manager
```
Comment (irappa-pattar) [inline lib-utilities/go.mod:12]: don't use local references

**Review by irappa-pattar — CHANGES_REQUESTED** [review]

File: lib-utilities/go.mod (hunk 5 above)
Comment (hemantha-beecherla) [reply lib-utilities/go.mod:12]: Local References will be removed, before merging to development in ODIM-Project branch

File: lib-persistence-manager/persistencemgr/redis.go (hunk 1 above)
Comment (hemantha-beecherla) [reply lib-persistence-manager/persistencemgr/redis.go]: Done

File: lib-persistence-manager/persistencemgr/redis.go (hunk 2 above)
Comment (hemantha-beecherla) [reply lib-persistence-manager/persistencemgr/redis.go]: Done

File: lib-persistence-manager/persistencemgr/redis.go (hunk 4 above)
Comment (hemantha-beecherla) [reply lib-persistence-manager/persistencemgr/redis.go]: done

File: lib-persistence-manager/persistencemgr/redis.go (hunk 2 above)
Comment (hemantha-beecherla) [reply lib-persistence-manager/persistencemgr/redis.go]: Done

File: lib-persistence-manager/persistencemgr/redis.go (hunk 3 above)
Comment (hemantha-beecherla) [reply lib-persistence-manager/persistencemgr/redis.go]: Done

**Review by irappa-pattar — APPROVED** [review]

**Pull request merged**